### DISCUSSION TITLE
Memory leak when training on Sagemaker p4 instances with distributed training

### DISCUSSION THREAD

**Discussion by ririya** [Lightning-AI/pytorch-lightning#19095] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi everyone, I've been trying to track down this issue for a few weeks now.

I'm training a semantic search model (MPNET) that generates embeddings for a pair of input strings.
 I'm also facing the same issue when training a cross encoder with the same inputs using Deberta-v3. 

For a few weeks I've been getting a memory leak only when training on Sagemaker p4 instances (8 A100 gpus), and only when using multiple instances. I've had problems before where I found out the issue was due to accumulating loss tensors in a dictionary, and solved the issue by calling loss.item(), so the gradients are not stored. 

However I'm still getting issues with this specific combination of PL + SMdistributed + 5 P4 instances. I believe the problem lies somewhere with the gradients not being released properly from memory. 

Anyone ever seen anything similar?

![image](https://github.com/Lightning-AI/pytorch-lightning/assets/23408930/179bb20c-a3a0-4504-90d2-86aa9710985d)
